src/util/assert.h

```diff
@@ -56,14 +56,16 @@ inline void mixxx_release_assert(const char* assertion, const char* file, int li
 #ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED
 #define DEBUG_ASSERT(cond)                                                  \
     do                                                                      \
-        if (Q_UNLIKELY(!static_cast<bool>(cond))) {                         \
+        if (!static_cast<bool>(cond)) [[unlikely]] {                        \
             mixxx_debug_assert(#cond, __FILE__, __LINE__, ASSERT_FUNCTION); \
         }                                                                   \
     while (0)
+#define MIXXX_RELEASE_NOEXCEPT
 #else
 #define DEBUG_ASSERT(cond) \
     do {                   \
     } while (0)
+#define MIXXX_RELEASE_NOEXCEPT noexcept
```

Comment:
What is the use for this? Since we don't use exception handling in Mixxx, does it at any value?

Reply:
The DEBUG_ASSERT macro, uses lots of external C++ code (Qt), which may throw, std::bad_alloc for example or maybe on I/O error when printing to the log I suppose. Unless you trust you assert macro to absolutely never throw?

Reply:
Indeed Qt doesn't throw exceptions so it was unnecessary. I remove it.